src/Game/Systems/BoostSystem.cpp

```diff
@@ -0,0 +1,63 @@
+#include "Systems/BoostSystem.h"
+
+BoostSystem::BoostSystem(SystemParams params)
+    : System(params)
+{
+    EVENT_SUBSCRIBE_MEMBER(m_EPlayerDamage, &BoostSystem::OnPlayerDamage);
+}
+
+bool BoostSystem::OnPlayerDamage(Events::PlayerDamage& e)
+{
+    if (e.Victim.ID == e.Inflictor.ID) {
+        return false;
+    }
+    if (!e.Victim.Valid() && e.Victim != LocalPlayer && !e.Victim.IsChildOf(LocalPlayer)) {
+        return false;
+    }
+    if (!e.Inflictor.Valid() || !e.Victim.Valid()) {
+        return false;
+    }
+
+    //if its not friendly fire, return
+    if ((int)m_World->GetComponent(e.Inflictor.ID, "Team")["Team"] != (int)m_World->GetComponent(e.Victim.ID, "Team")["Team"]) {
+        return false;
+    }
+
+    //determine the inflictors class
+    auto className = DetermineClass(e.Inflictor);
+    if (className == "") {
+        return false;
+    }
+
+    //get the XML file, example: "Schema/Entities/BoostclassName.xml"
+    std::string classXML = "Schema/Entities/" + className + ".xml";
```

Comment:
Since this is only 3 different files, is it not better to cach them instead of loading them every time a player gets shot?